android/src/main/java/com/twiliovoicereactnative/VoiceBroadcastReceiver.java

```diff
@@ -128,6 +130,18 @@ public void onReceive(Context context, Intent intent) {
         params.putMap(EVENT_KEY_CALL_INVITE_INFO, callInviteInfo);
 
         AndroidEventEmitter.getInstance().sendEvent(ScopeVoice, params);
+
+        if (Storage.callAcceptedPromiseMap.containsKey(uuid)) {
+          Promise promise = Storage.callAcceptedPromiseMap.get(uuid);
+          final Call call = Storage.callMap.get(uuid);
+          final WritableMap callInfo = serializeCall(uuid, call);
+
+          Storage.releaseCallInviteStorage(uuid, callInvite.getCallSid(), Storage.uuidNotificaionIdMap.get(uuid), "accept");
+
+          promise.resolve(callInfo);
+        } else {
+          Storage.releaseCallInviteStorage(uuid, callInvite.getCallSid(), Storage.uuidNotificaionIdMap.get(uuid), "accept");
```

Comment:
Is this point in the code the final failstate? Should we reject the promise here?

Reply:
Not all 'accept' call flows are paired with a promise object, for example, when a notification action/button is selected.

Reply:
Ah I see. None of those functions can throw right? I'm concerned if there's an edge case such that the Promise is never resolved or rejected, and the JS layer hangs.

Reply:
its low probability, but good call out.. its fixed now.